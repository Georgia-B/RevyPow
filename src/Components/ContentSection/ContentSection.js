import React from 'react';

const ContentSection = (props) => {
    return (
        <section className={props.className ? `content-section ${props.className}` : "content-section"}>
            <img className="content-section__icon" src={props.image} alt={props.alt} />
            {props.children}
            <div className="content-section__spacer"></div>
        </section>
    );
}

export default ContentSection;
