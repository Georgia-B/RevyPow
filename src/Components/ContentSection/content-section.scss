.content-section {
    display: flex;
    width: 80%;
    justify-content: space-between;
    align-items: center;
    margin: 10px 0;

    &__spacer,
    &__icon {
        width: 50px;
    }
}

.weather-container {
    align-items: center;
}

@media (max-width: $breakpoint-large) {
    .content-section {
        width: 95%;
    }
}

@media (max-width: $breakpoint-medium) {
    .content-section {
        justify-content: center;

        &__icon,
        &__spacer {
            display: none;
        }
    }
    .temperature-container {
        justify-content: space-between;
    }
}

@media (max-width: $breakpoint-small) {
    .content-section {
        width: 100%;
    }
}