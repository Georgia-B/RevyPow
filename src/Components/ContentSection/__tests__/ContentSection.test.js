import React from 'react'
import { cleanup, render } from '@testing-library/react'
import ContentSection from '../ContentSection';

describe('Components | <ContentSection /> with custom class', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(
            <ContentSection className="testclass">
                <h1>Hello World</h1>
            </ContentSection>);
        expect(component.asFragment()).toMatchSnapshot();
    })
})

describe('Components | <ContentSection /> with no class', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(
            <ContentSection>
                <h1>Hello World</h1>
            </ContentSection>);
        expect(component.asFragment()).toMatchSnapshot();
    })
})