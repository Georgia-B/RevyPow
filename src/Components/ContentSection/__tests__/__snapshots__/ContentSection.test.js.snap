// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <ContentSection /> with custom class should render correctly 1`] = `
<DocumentFragment>
  <section
    class="content-section testclass"
  >
    <h1>
      Hello World
    </h1>
  </section>
</DocumentFragment>
`;

exports[`Components | <ContentSection /> with no class should render correctly 1`] = `
<DocumentFragment>
  <section
    class="content-section"
  >
    <h1>
      Hello World
    </h1>
  </section>
</DocumentFragment>
`;
